### ISSUE TITLE
Question on Belebele Evaluation Protocol for Marco-Mini-Base / Marco-Nano-Base (Table 18)

### ISSUE THREAD

**Issue by yanliang3612** [AIDC-AI/Marco-Bench#5]
Dear Marco-MoE Team,

We are conducting research based on your Marco-MoE base models, using Belebele (Table 18 in your paper) as our primary benchmark. We are reaching out because we have been unable to fully reproduce the reported Belebele scores—particularly for Marco-Nano-Base—and would like to confirm whether our evaluation setup matches yours.

**1. Our evaluation setup (zero-shot)**

We evaluate the HuggingFace checkpoints AIDC-AI/Marco-Mini-Base and AIDC-AI/Marco-Nano-Base with lm-evaluation-harness (lm_eval==0.4.10), using the following configuration:

Tasks: belebele_{lang} for all 29 pretraining languages (e.g., belebele_eng_Latn, belebele_arb_Arab)

Few-shot: --num_fewshot 0 (strictly zero-shot; no in-context examples)

Dataset: facebook/belebele (standard test split)

Scoring: multiple-choice via log-likelihood over answer choices (not free-form generation)

**Prompt template (lm_eval default for Belebele):**
```bash

      P: {passage}
      
      Q: {question}
      
      A: {option_1}
      
      B: {option_2}
      
      C: {option_3}
      
      D: {option_4}
```

**Answer:**

```bash
    No chat template or system prompt (chat_template: null)
    
    Model arguments: trust_remote_code=True, dtype=bfloat16
    
    Metric: accuracy (acc), averaged over the 29 pretraining languages

The same pipeline is used for both Marco-Mini-Base and Marco-Nano-Base.
```

**2. Discrepancy with Table 18**

Our reproduced scores differ from the numbers reported in Table 18. The gap is relatively modest for Marco-Mini-Base, but substantially larger and fairly consistent across languages for Marco-Nano-Base. Detailed per-language comparisons are provided in the attached screenshot(s). We suspect this may stem from differences in the evaluation protocol rather than model weights alone, but we would like to verify this with you before drawing conclusions.

**3. Questions**

  - Could you please clarify how Belebele was evaluated for Table 18?
  
  - Which framework or script did you use (e.g., lm_eval, a custom evaluation script, or another harness)?
  
  - What was the exact prompt template? Was a chat template applied?
  
  - Was the evaluation strictly zero-shot, as in our setup?
  
  - Did you use log-likelihood over choices A/B/C/D, or generation with post-hoc answer parsing?
  
  - Which model checkpoint or revision was evaluated (HuggingFace main vs. an internal checkpoint)?
  
  - Were there any other relevant settings (e.g., lm_eval version, dtype, length normalization)?

We would greatly appreciate any details or evaluation scripts you are able to share. We are also happy to provide our evaluation scripts, result JSON files (which record the full task configuration), or additional breakdowns if helpful.

Thank you for open-sourcing Marco-MoE and for your time.

<img width="779" height="803" alt="Image" src="https://github.com/user-attachments/assets/97b5d807-6499-46e5-a4c1-f9b508bb6061" />

**Comment by Sniper970119** [comment]
Hi, @yanliang3612 ,

Thank you for your interest in Marco-LLM and for the detailed write-up — we really appreciate the effort you put into reproducing our results.

We are currently cleaning up our evaluation scripts and will release them shortly. In the meantime, here are the key details to help you align your setup with ours:

Framework: We use[LightEval](https://github.com/huggingface/lighteval)(v0.10.0) with custom task definitions, not `lm-evaluation-harness`. This is likely the primary source of the discrepancy.

Prompt formulation: We use LightEval's `MCFFormulation()` (Multiple Choice Formulation) via `get_mcq_prompt_function`. Importantly, the instructional framing is in English (`eng_Latn`) for all language subsets — i.e., structural tokens like "Question:", "Answer:", and choice prefixes (A/B/C/D) are in English, while the passage and question content remain in the target language. This prompt template differs from the default Belebele template in `lm-evaluation-harness`, and is likely the core reason for the score gap.

Scoring: `loglikelihood_acc_metric(normalization=None)` — we compare the log-likelihood of each answer choice (A/B/C/D) and select the one with the highest probability. No normalization is applied. This should be consistent with your setup.

Few-shot: Strictly zero-shot (0 few-shot examples), same as yours.

Dataset: `facebook/belebele`, `test` split — same as yours.

Model checkpoints: The HuggingFace checkpoints (`AIDC-AI/Marco-Mini-Base` and `AIDC-AI/Marco-Nano-Base`) are the exact ones we used. dtype is `bfloat16`.

We believe the prompt template difference is the main factor. Different frameworks construct prompts differently (choice formatting, instruction wording, etc.), and base models — especially smaller ones like Marco-Nano — tend to be quite sensitive to prompt format variations.

We will release the full evaluation code soon. In the meantime, if you are still unable to reproduce results at a comparable level after aligning the setup above, please feel free to follow up here and we will prioritize releasing the exact evaluation scripts.

Best regards

**Comment by yanliang3612** [comment]
Thanks. I will have a try first and follow up here.

**Comment by yanliang3612** [comment]
Hi @Sniper970119 ,

Thank you again for the detailed clarification on the Belebele evaluation setup — the LightEval + MCFFormulation() protocol with English structural framing was very helpful, and we have started aligning our pipeline accordingly.

That said, we are still unable to fully reproduce many of the reported results in Table 18 (especially for Marco-Nano-Base, where the gap remains substantial even after switching away from lm-evaluation-harness). Without the exact evaluation code, it is difficult for us to rule out subtle differences in prompt formatting, task configuration, or other implementation details.

Would it be possible for the team to release the full Belebele evaluation scripts at your earliest convenience? Having the official code would greatly help the community verify and build on your results. We are happy to re-run everything on our side and share updated numbers once it is available.

Thanks again for open-sourcing Marco-MoE and for taking the time to help with this.

**Comment by yanliang3612** [comment]
Hi @Sniper970119 , 

Just following up on my previous comment. I wanted to check whether there are any updates on releasing the full Belebele evaluation scripts, or at least the exact prompts/configuration used for the Table 18 results.

We are still unable to fully reproduce several of the reported numbers, especially for Marco-Nano-Base, and having the official evaluation code would be very helpful for ruling out differences in prompt formatting, task setup, or other implementation details.

A rough timeline or any pointer to the relevant scripts/configs would be greatly appreciated. Thanks again for your help and for open-sourcing Marco-MoE.

**Comment by Sniper970119** [comment]
Hi, @yanliang3612 

Sorry for the late reply — I was on vacation and didn't notice this email in time.

Our relevant inference code has been uploaded to:
🔗 https://github.com/Sniper970119/Marco-LLM-eval/

Additionally, we have provided some inference details for Belebele at:
🔗 https://github.com/Sniper970119/Marco-LLM-eval/tree/main/eval_results

We hope this will be helpful to you.

Please note that the code was organized with the assistance of AI (primarily to remove sensitive information such as file paths and internal naming conventions). Some of my colleagues will verify additional details later, but we wanted to address your concerns as quickly as possible, so we uploaded this portion of the code in advance. Feel free to give it a try. In particular, the logit information is provided, which should be especially useful for reproduction.

If you need more detailed disclosures or results from additional datasets, please don't hesitate to let us know. We will organize and upload the relevant results accordingly.

Best regards

**Comment by yanliang3612** [comment]
Thank you very much!
